platforms/core-runtime/base-services/src/main/java/org/gradle/util/internal/CollectionUtils.java

```diff
@@ -136,6 +136,9 @@ public static <T> List<T> filter(T[] array, Spec<? super T> filter) {
         return filter(Arrays.asList(array), Lists.<T>newArrayListWithCapacity(array.length), filter);
     }
 
+    public static <T> List<T> filterIndexed(List<? extends T> list, IndexedSpec<? super T> filter) {
+        return filterIndexed(list, Lists.<T>newArrayListWithCapacity(list.size()), filter);
```

Comment:
🤔 Would it be better to produce immutable lists instead? Are these copied downstream?